sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/aggregates/AggregateCompiler.java

```diff
@@ -379,10 +440,32 @@ void processArrayAgg(SqlBasicAggFunction function) {
         DBSPTypeUser semigroup = new DBSPTypeUser(
                 node, SEMIGROUP, "ConcatSemigroup", false, accumulatorType);
         DBSPVariablePath p = accumulatorType.var();
-        String convertName = "to_array";
-        if (arrayType.mayBeNull)
-            convertName += "N";
-        DBSPClosureExpression post = new DBSPApplyExpression(convertName, arrayType, p).closure(p);
+        DBSPClosureExpression post;
+        if (collations.isEmpty()) {
+            String convertName = "to_array";
+            if (arrayType.mayBeNull)
+                convertName += "N";
+            post = new DBSPApplyExpression(convertName, arrayType, p).closure(p);
+        } else {
+            String convertName = "sort";
```

Comment:
lets hope we never change the name

Reply:
the whole sqllibJava works like this.
I may choose a longer name.